lib/garage/restful_actions.rb

```diff
@@ -32,7 +32,22 @@ def resource_class=(klass)
       end
 
       def resource_class
-        @resource_class ||= name.sub(/Controller\z/, '').demodulize.singularize.constantize
+        @resource_class ||= resource_class_name.constantize
+      end
+
+      private
+
+      def resource_class_name
+        class_name = name.sub(/Controller\z/, '').demodulize.singularize
+        class_name_with_suffix = "#{class_name}Resource"
+        case
+        when const_defined?(class_name_with_suffix)
```

Comment:
(YAGNI?) I may want to customize naming convention not `FooResource`  eg. `FooPresentation`.

I propose to extract this resource class resolver like this:

```ruby
class Garage::ResourceClassResolver
  def initialize(&block)
    @block = block
  end

  def call(controller_class_name)
    class_name = class_name.sub(/Controller\z/, '').demodulize.singularize
    
    if const_defined?(@blocl.call(class_name)
      ...
    elsif const_defined?(class_name)
      ...
    else
      ...
    end
  end
end

# defalut initalization
Garage.configuration.resouce_class_resolver =
  Garage::ResourceClassResolver.new {|class_name| "#{class_name}Resource" }

# in RestfulActions
def resource_class
  Garage.configuration.resouce_class_resolver.call(name)
end
```
Then I can customize lookup pattern by setting `Garage.configuration.resouce_class_resolver =` in my `config/initializers` .

How do you think about this?